tests/imbu/unit/imbu_models_test.py

```diff
@@ -311,56 +323,62 @@ def testMergeRanges(self):
     """ Tests the mergeRanges() method used in fragmenting Imbu results."""
     imbu = self._setupFakeImbuModelsInstance()
 
-    def mergeIntoList(ranges):
-      return list(imbu._mergeRanges(ranges))
-
-    ranges = [(3, 7), (3, 5), (0, 4)]
-    self.assertItemsEqual([(0, 7)], mergeIntoList(ranges))
-    ranges = [(5, 6), (3, 4), (1, 2)]
-    self.assertItemsEqual([(1, 2), (3, 4), (5, 6)], mergeIntoList(ranges))
-    ranges = [(0, 13)]
-    self.assertItemsEqual(ranges, mergeIntoList(ranges))
-    ranges = [(0, 3), (6, 3)]
+    ranges = [[3, 7], [3, 5], [0, 4]]
+    self.assertItemsEqual([[0, 7]], list(imbu._mergeRanges(ranges)))
```

Comment:
CC @subutai @scottpurdy 

@oxtopus: I just learned something, and it doesn't bode well at all for our current use of assertItemsEqual in all our python projects, which turns out to be incredibly poorly named. It's so order agnostic, that in fact it doesn't even compare the elements, **only element counts**! They renamed it in Python 3 to `assertCountsEqual`, which makes more sense.

```
assertItemsEqual(self, expected_seq, actual_seq, msg=None) unbound unittest.case.TestCase method
    An unordered sequence specific comparison. It asserts that
    actual_seq and expected_seq have the same element counts.
```

I grep'ed some of our repos, and it looks like we have some cleanup work to do. I hope that we don't have buggy code, since all these assertions were actually only validating the counts, not the elements:

```
vkruglikov@Vitalys-MacBook-Pro:~/nta/nupic (master)$ git grep assertItemsEqual
tests/unit/nupic/algorithms/knn_classifier_test.py:    self.assertItemsEqual(classifier.getPatternIndicesWithPartitionId(433),
tests/unit/nupic/algorithms/knn_classifier_test.py:    self.assertItemsEqual(classifier.getPatternIndicesWithPartitionId(666),
tests/unit/nupic/algorithms/knn_classifier_test.py:    self.assertItemsEqual(classifier.getPatternIndicesWithPartitionId(413),
tests/unit/nupic/algorithms/knn_classifier_test.py:    self.assertItemsEqual(classifier.getPartitionIdPerPattern(),
tests/unit/nupic/algorithms/knn_classifier_test.py:    self.assertItemsEqual(classifier.getPartitionIdList(),[433, 413, 213])
tests/unit/nupic/algorithms/knn_classifier_test.py:    self.assertItemsEqual(classifier.getPatternIndicesWithPartitionId(433),
tests/unit/nupic/algorithms/knn_classifier_test.py:    self.assertItemsEqual(classifier.getPatternIndicesWithPartitionId(413),
tests/unit/nupic/algorithms/knn_classifier_test.py:    self.assertItemsEqual(classifier.getPartitionIdPerPattern(), [433, 413])
tests/unit/nupic/algorithms/knn_classifier_test.py:    self.assertItemsEqual(classifier.getPartitionIdList(),[433, 413])
vkruglikov@Vitalys-MacBook-Pro:~/nta/nupic (master)$ cd ../nupic.core/
vkruglikov@Vitalys-MacBook-Pro:~/nta/nupic.core (master)$ git grep assertItemsEqual
vkruglikov@Vitalys-MacBook-Pro:~/nta/nupic.core (master)$ cd ../numenta-apps/
vkruglikov@Vitalys-MacBook-Pro:~/nta/numenta-apps (master)$ git grep assertItemsEqual
htm.it/tests/py/integration/app/adapters/datasource/cloudwatch/cw_datasource_adapter_base_test.py:    self.assertItemsEqual(regionNames, expectedRegionNames)
htm.it/tests/py/integration/app/adapters/datasource/cloudwatch/cw_datasource_adapter_base_test.py:    self.assertItemsEqual(supportedResourceTypes, self._expectedResourceTypes)
htm.it/tests/py/integration/app/adapters/datasource/cloudwatch/cw_datasource_adapter_base_test.py:    self.assertItemsEqual(
htm.it/tests/py/integration/app/adapters/datasource/cloudwatch/cw_datasource_adapter_base_test.py:        self.assertItemsEqual(metricInfo.keys(),
htm.it/tests/py/integration/app/adapters/datasource/datasource_adapter_factory_test.py:    self.assertItemsEqual(datasource_adapter_factory.listDatasourceNames(),
htm.it/tests/py/integration/app/runtime/aggregator_metric_collection_test.py:      self.assertItemsEqual(set(timestamps), timestamps)
htm.it/tests/py/integration/app/runtime/aggregator_metric_collection_test.py:      self.assertItemsEqual(instances, set(instances))
htm.it/tests/py/integration/app/webservices/annotations_api_test.py:    self.assertItemsEqual([expectedAnnotation], actual)
htm.it/tests/py/integration/app/webservices/annotations_api_test.py:    self.assertItemsEqual([expectedAnnotation], actual)
htm.it/tests/py/integration/app/webservices/annotations_api_test.py:    self.assertItemsEqual([expectedAnnotation], actual)
htm.it/tests/py/integration/app/webservices/cloudwatch_api_test.py:      self.assertItemsEqual(res.keys(), attributes)
htm.it/tests/py/integration/app/webservices/cloudwatch_api_test.py:      self.assertItemsEqual(res.keys(), attributes)
htm.it/tests/py/integration/app/webservices/instances_api_test.py:    self.assertItemsEqual(initialGetResult, [])
htm.it/tests/py/integration/app/webservices/instances_api_test.py:    self.assertItemsEqual(getIntialResult, [])
htm.it/tests/py/integration/app/webservices/instances_api_test.py:    self.assertItemsEqual([instanceId.rpartition("/")[2]
htm.it/tests/py/integration/app/webservices/instances_api_test.py:    self.assertItemsEqual(getPostDeleteResult, [])
htm.it/tests/py/integration/app/webservices/models_api_test.py:    self.assertItemsEqual(postResult.keys(),
htm.it/tests/py/integration/app/webservices/models_api_test.py:    self.assertItemsEqual(getModelResult[0].keys(),
htm.it/tests/py/integration/app/webservices/models_api_test.py:    self.assertItemsEqual(allModelsResult[0].keys(),
htm.it/tests/py/integration/app/webservices/models_api_test.py:    self.assertItemsEqual(allModelsResult[0].keys(),
htm.it/tests/py/integration/app/webservices/models_api_test.py:    self.assertItemsEqual(httpData.keys(), ["metrics", "names"])
htm.it/tests/py/integration/app/webservices/models_api_test.py:    self.assertItemsEqual(httpData["names"], ["timestamp",
htm.it/tests/py/integration/app/webservices/models_api_test.py:    self.assertItemsEqual(httpsData.keys(), ["metrics", "names"])
htm.it/tests/py/integration/app/webservices/models_api_test.py:    self.assertItemsEqual(httpsData["names"], ["timestamp",
htm.it/tests/py/integration/app/webservices/models_api_test.py:    self.assertItemsEqual(getModelResult[0].keys(),
htm.it/tests/py/integration/app/webservices/models_api_test.py:    self.assertItemsEqual(allModelsResult[0].keys(),
htm.it/tests/py/integration/app/webservices/models_api_test.py:    self.assertItemsEqual(allModelsResult[0].keys(),
htm.it/tests/py/integration/app/webservices/models_api_test.py:    self.assertItemsEqual(getMetricDataResult.keys(), ["data", "names"])
htm.it/tests/py/integration/app/webservices/models_api_test.py:    self.assertItemsEqual(getMetricDataWithAnomalyQueryResult.keys(),
htm.it/tests/py/integration/app/webservices/models_api_test.py:    self.assertItemsEqual(getAllMetricDataResult.keys(), ["data", "names"])
htm.it/tests/py/integration/app/webservices/models_api_test.py:    self.assertItemsEqual(httpsData.keys(), ["data", "names"])
htm.it/tests/py/integration/app/webservices/models_api_test.py:    self.assertItemsEqual(httpsData["names"], ["timestamp",
htm.it/tests/py/integration/app/webservices/models_api_test.py:    self.assertItemsEqual(httpData.keys(), ["data", "names"])
htm.it/tests/py/integration/app/webservices/models_api_test.py:    self.assertItemsEqual(httpData["names"], ["timestamp",
htm.it/tests/py/unit/app/webservices/cloudwatch_api_test.py:    self.assertItemsEqual(result,
htmengine/tests/integration/runtime/metric_garbage_collector_test.py:      self.assertItemsEqual(
htmengine/tests/unit/model_checkpoint_mgr/model_checkpoint_mgr_test.py:    self.assertItemsEqual(ids, expModelIDs)
taurus_engine/tests/unit/check_model_consistency_test.py:    self.assertItemsEqual(warnings, allExpectedWarnings)
taurus_engine/tests/unit/check_model_consistency_test.py:    self.assertItemsEqual(errors, allExpectedErrors)
taurus_engine/tests/unit/reset_all_taurus_engine_data_test.py:    self.assertItemsEqual(expectedDeleteExchangeCalls,
taurus_metric_collectors/tests/integration/purge_old_tweets_test.py:      self.assertItemsEqual(
taurus_metric_collectors/tests/unit/delete_companies_test.py:    self.assertItemsEqual(deleteMetricMock.call_args_list,
taurus_metric_collectors/tests/unit/delete_companies_test.py:    self.assertItemsEqual(deleteMetricMock.call_args_list,
taurus_metric_collectors/tests/unit/metric_maintenance_agent_test.py:    self.assertItemsEqual(expected, readyMetricNames)
taurus_metric_collectors/tests/unit/metric_maintenance_agent_test.py:    self.assertItemsEqual(expected, readyMetricNames)
taurus_metric_collectors/tests/unit/metric_maintenance_agent_test.py:    self.assertItemsEqual(expected,
taurus_metric_collectors/tests/unit/metric_maintenance_agent_test.py:    self.assertItemsEqual(["foo", "bar"], symbols)
taurus_metric_collectors/tests/unit/metric_utils_test.py:    self.assertItemsEqual(expected, metricNames)
taurus_metric_collectors/tests/unit/metric_utils_test.py:    self.assertItemsEqual(subsetOfMetricNames,
taurus_metric_collectors/tests/unit/metric_utils_test.py:    self.assertItemsEqual(positives, filteredNames)

```